### PR TITLE
Add CONFIG_MCUBOOT_BOOTLOADER_NO_DOWNGRADE to MCUboot module

### PR DESCRIPTION
**Pull request by de-nordic**

### CODE REVIEW

**Review by nordicjm — COMMENTED** [review]
What would be good for this is if an option was added to sysbuild so that it could enable the option in MCUboot automatically

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -353,6 +353,11 @@ HALs
 MCUboot
 *******
 
+  * Added :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_WITH_DOWNGRADE_PREVENTION`
+    that allows to inform application that the on-board MCUboot has been configured
+    with download prevention enabled. This option is automatically selected for
```
[the remark this answers is not in the document]
Comment (de-nordic) [reply doc/releases/release-notes-3.5.rst:358]: Fixed

File: modules/Kconfig.mcuboot [hunk 2]
```diff
@@ -154,30 +156,60 @@ config MCUBOOT_BOOTLOADER_MODE_SINGLE_APP
...
+	  MCUBOOT_BOOTLOADER_MODE_DOWNGRADE_PREVENTION as it is not possible
+	  to swap back to older version of application.
 
 endchoice # MCUBOOT_BOOTLOADER_MODE
 
+config MCUBOOT_BOOTLOADER_MODE_SUPPORTS_DOWNGRADE_PREVENTION
+	bool
+	help
+	  Selected mode supports downgrade prevention, where you can not switch to
+	  Selected mode supports downgrade prevention, where you cannot switch to
```
Comment (nordicjm) [inline modules/Kconfig.mcuboot:198]: Duplicated line

File: doc/releases/release-notes-3.5.rst [hunk 3]
```diff
@@ -353,6 +353,11 @@ HALs
 MCUboot
 *******
 
+  * Added :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_WITH_DOWNGRADE_PREVENTION`
+    that allows to inform application that the on-board MCUboot has been configured
+    with downgrade  prevention enabled. This option is automatically selected for
```
Comment (nordicjm) [inline doc/releases/release-notes-3.5.rst:358]: double space

File: modules/Kconfig.mcuboot (hunk 2 above)
Comment (de-nordic) [reply modules/Kconfig.mcuboot:197]: Fixed

File: doc/releases/release-notes-3.5.rst (hunk 3 above)
Comment (de-nordic) [reply doc/releases/release-notes-3.5.rst:358]: Fixed

File: modules/Kconfig.mcuboot [hunk 4]
```diff
@@ -154,30 +156,59 @@ config MCUBOOT_BOOTLOADER_MODE_SINGLE_APP
...
 	  select one with higher application image version, which usually
 	  means major.minor.patch triple, unless BOOT_VERSION_CMP_USE_BUILD_NUMBER
 	  is also selected that enables comparison of build number.
+	  This option automatically selectes
+	  MCUBOOT_BOOTLOADER_MODE_DOWNGRADE_PREVENTION as it is not possible
+	  to swap back to older version of application.
 
 endchoice # MCUBOOT_BOOTLOADER_MODE
 
+config MCUBOOT_BOOTLOADER_MODE_SUPPORTS_DOWNGRADE_PREVENTION
```
Comment (carlescufi) [inline modules/Kconfig.mcuboot:194]: `MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE`

File: modules/Kconfig.mcuboot [hunk 5]
```diff
@@ -154,30 +156,59 @@ config MCUBOOT_BOOTLOADER_MODE_SINGLE_APP
...
+	bool
+	help
+	  Selected mode supports downgrade prevention, where you cannot switch to
+	  an application with lower version than the currently running application.
+
+if MCUBOOT_BOOTLOADER_MODE_SUPPORTS_DOWNGRADE_PREVENTION
+config MCUBOOT_BOOTLOADER_MODE_WITH_DOWNGRADE_PREVENTION
```
Comment (carlescufi) [inline modules/Kconfig.mcuboot:201]: `MCUBOOT_BOOTLOADER_NO_DOWNGRADE`

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**